src/components/sync-dialog/SyncDialog.jsx

```diff
@@ -41,20 +40,25 @@ class SyncDialog extends React.Component {
     };
 
     handleSynchronization = async () => {
-        this.props.loading.show(true, i18n.t("Synchronizing metadata"));
+        const { handleClose, loading, metadata, d2 } = this.props;
+        loading.show(true, i18n.t("Synchronizing metadata"));
 
         try {
-            const syncReport = await startSynchronization(this.props.d2, {
-                metadata: this.props.metadata,
+            const builder = {
+                metadata: metadata,
                 targetInstances: this.state.targetInstances,
-            });
-            this.props.handleClose(syncReport);
+            };
+            for await (const { message, syncReport, done } of startSynchronization(d2, builder)) {
+                if (message) loading.show(true, message);
+                if (syncReport) await syncReport.save();
+                if (done) handleClose(syncReport);
```

Comment:
Minor: I guess `done` is implicit to the generator stopping, could we simply do the handleClose out of the for loop? Or is there a case where you yield done=true and later yield other values?

Reply:
I added this inside so we do not have to store the last syncReport in a temporal local variable outside the for...of.

```
let lastSyncReport = undefined;
for await (const { message, syncReport, done } of startSynchronization(d2, builder)) {
                if (message) loading.show(true, message);
                if (syncReport) await syncReport.save();
                lastSyncReport = syncReport;
handleClose(lastSyncReport);
```

Also note that this done is not the implicit done of the Generator, during the for...of you loose the internal properties and only receive the value.

Reply:
Yes, that's a custom `done` key in the object. Ok, I see your point.